src/extension-test/unit/cursor-doc/paredit-test.ts

```diff
@@ -1058,6 +1058,53 @@ describe('paredit', () => {
                 paredit.addRichComment(a);
                 expect(textAndSelection(a)).toEqual(textAndSelection(b));
             });
-        })
+        });
+
+        describe('splice sexp', () => {
+            it('splice empty', () => {
+                const a = docFromTextNotation('|');
+                paredit.spliceSexp(a);
+                expect(text(a)).toEqual('');
+            });
+
+            it('splice list', () => {
+                const a = docFromTextNotation('(a| b c)');
+                paredit.spliceSexp(a);
+                expect(text(a)).toEqual('a b c');
+            });
+
+            it('splice vector', () => {
+                const a = docFromTextNotation('[a| b c]');
+                paredit.spliceSexp(a);
+                expect(text(a)).toEqual('a b c');
+            });
+
+            it('splice map', () => {
+                const a = docFromTextNotation('{a| b}');
+                paredit.spliceSexp(a);
+                expect(text(a)).toEqual('a b');
+            });
+
+
+            it('splice nested', () => {
+                const a = docFromTextNotation('[1 {ab| cd} 2]');
+                paredit.spliceSexp(a);
+                expect(text(a)).toEqual('[1 ab cd 2]');
+            });
+
+            // TODO: enable after fixing spliceSexp
+            xit('splice set', () => {
+                const a = docFromTextNotation('#{a| b}');
+                paredit.spliceSexp(a);
+                expect(text(a)).toEqual('a b');
+            });
+
+            // TODO: enabling this breaks bunch of other tests. Not sure why
```

Comment:
Maybe you guys can clue me in on why enabling this test causes other tests to break. So puzzled!

Reply:
I don't know. There is another TODO in `src/extension-test/unit/cursor-doc/token-cursor-test.ts` where I encountered a similar thing. 

This is stuff for nightmares, really. The token cursor is a very mutating creature. But this that it leaks between documents is crazy.